src/sql/workbench/contrib/notebook/test/browser/notebookEditor.test.ts

```diff
@@ -155,19 +194,230 @@
 		assert.strictEqual(currentPosition, selectedRange, 'notebookEditor.getPosition() should return the range that was selected');
 	});
 
-	// NotebookEditor-getCellEditor tests.
 	['', undefined, null, 'unknown string', /*unknown guid*/generateUuid()].forEach(input => {
-		test(`NotebookEditor: Negative Test for getCellEditor() - returns undefined for invalid or unknown guid:'${input}'`, async () => {
-			const result = notebookEditor.getCellEditor(input);
-			assert.strictEqual(result, undefined, `notebookEditor.getCellEditor() should return undefined when invalid guid:'${input}' is passed in for a notebookEditor of an empty document.`);
+		test(`NotebookEditor: Negative Test -> getCellEditor() returns undefined for invalid or unknown guid:'${input}'`, async () => {
+			await setupNotebookEditor(notebookEditor, untitledNotebookInput);
+			const inputGuid = <string>input;
+			const result = notebookEditor.getCellEditor(inputGuid);
+			assert.strictEqual(result, undefined, `notebookEditor.getCellEditor() should return undefined when invalid guid:'${inputGuid}' is passed in for a notebookEditor of an empty document.`);
 		});
 	});
 
-	test('NotebookEditor: Positive Test for getCellEditor() - returns a valid text editor object for valid guid input', async () => {
+	test('NotebookEditor: Positive Test -> getCellEditor() returns a valid text editor object for valid guid input', async () => {
+		await setupNotebookEditor(notebookEditor, untitledNotebookInput);
 		const result = notebookEditor.getCellEditor(cellTextEditorGuid);
 		assert.strictEqual(result, queryTextEditor, 'notebookEditor.getCellEditor() should return an expected QueryTextEditor when a guid corresponding to that editor is passed in.');
 
 	});
 
+	test('NotebookEditor: Positive Test -> verifies that domNode passed in via addOverlayWidget() call gets attached to the root HtmlElement of notebookEditor', async () => {
+		await setupNotebookEditor(notebookEditor, untitledNotebookInput);
+		const domNode: HTMLElement = document.createElement('div');
+		const widget: IOverlayWidget = {
+			getId(): string { return '' },
+			getDomNode(): HTMLElement { return domNode; },
+			getPosition(): IOverlayWidgetPosition | null { return null; }
+		}
+		notebookEditor.addOverlayWidget(widget);
+		const rootElement: HTMLElement = notebookEditor['_overlay'];
+		assert.ok(domNode.parentElement === rootElement, `parent of the passed in domNode must be the root element of notebookEditor:${notebookEditor}`);
+	});
+
+	test('NotebookEditor: Noop methods do not throw', async () => {
```

Comment:
It is debatable that no-op methods are worth testing for, verifying that they do not throw is probably good enough.